game.rb

```diff
@@ -0,0 +1,58 @@
+require_relative 'game'
+
+class Game
+  attr_reader :won, :code
+
+  def initialize(code = nil)
+    @colors = %w[red blue green orange yellow brown]
+    @code = code || 4.times.map { @colors.sample }
+    @won = false
+  end
+
+  def take_input
+    puts 'Please enter your guess'
+    guess = gets.chomp.downcase
+    until valid?(guess)
+      puts 'Please enter a valid guess'
+      guess = gets.chomp.downcase
+    end
+    guess
+  end
+
+  def process_guess(guess)
```

Comment:
Nitpick: I would include this in `take_input`. Make it return something that rest of the app can use right away.